### ISSUE TITLE
Add new transport option `state`

### ISSUE THREAD

**Issue by remcohaszing** [simov/grant#148]
Koa is based on middleware. This means middleware can run after the Grant middleware to process the response data instead of a callback URL. I suggest to add a new `transport` option named `state`. This makes Grant assign the response data to `ctx.state.grant.response` instead of `ctx.body`.

Current usage:

```js
import qs from 'qs';

export default async (ctx, next) => {
  // Runs grant middleware;
  await next();

  const response = qs.parse(ctx.body);
}
```

New usage:

```js
export default async (ctx, next) => {
  // Runs grant middleware;
  await next();

  const { response } = ctx.state.grant;
}
```

**Comment by simov** [comment]
Currently the following is possible using the `transport: 'session'` option:

```js
var app = new Koa()
app.keys = ['grant']
app.use(session(app))
app.use(async (ctx, next) => {
  await next()
  console.log(ctx.session.grant.response)
  ctx.body = 'foo'
})
app.use(grant(config))
koaqs(app)
app.listen(3000)
```

It still redirects to whatever your `callback` path is set to, but at least you have the response data inside the session instead.

**Comment by simov** [comment]
Transport `state` landed in v4.7.0

The documentation and examples are a bit scarce at the moment, but you should be able to figure out how to use it.

Thanks for the idea!

Here is the relevant commit in the [readme](https://github.com/simov/grant/commit/3b04eb69a278165ae9be7ba7a06e8b85da21c5e5).

Let me know if you have any issues.
